### ISSUE TITLE
Parameters: { evals_result } might not be used.

### ISSUE THREAD

**Issue by zhouquan03** [dmlc/xgboost#5692]
environment：
macOS 10.15.4

$ python -V
Python 3.7.7

$ pip freeze
...
xgboost==1.1.0
...

code:

    evals = ['auc', 'logloss']
    eval_set = [(train_x, train_y), (test_x, test_y)]
    
    model = XGBClassifier(
        learning_rate=0.1,
        n_estimators=30,
        max_depth=5,
        min_child_weight=1,
        gamma=0,
        subsample=0.8,
        colsample_bytree=0.8,
        objective='binary:logistic',
        nthread=8,
        scale_pos_weight=1,
        seed=1024,
        evals_result=evals
    )

    model.fit(train_x, train_y, eval_metric=evals, eval_set=eval_set)

warning message：
[10:30:44] WARNING: /Users/travis/build/dmlc/xgboost/src/learner.cc:480:
Parameters: { evals_result } might not be used.

  This may not be accurate due to some parameters are only used in language bindings but
  passed down to XGBoost core.  Or some parameters are not used but slip through this
  verification. Please open an issue if you find above cases.

**Comment by trivialfis** [comment]
I believe this is a correct message.  Passing `evals` for `eval_metric` is sufficient.  There's no `evals_result` on scikit learn interface.
